exec/exec.go

```diff
@@ -68,29 +68,38 @@ func (d *Exec) Run(tableName string, keyName string, command string, sleepStartR
 		dynamolock.FailIfLocked(),
 	)
 	if err != nil {
-		return err
+		// dynamolock.LockNotGrantedError doesn't export the error so this is the next best thing we can do.
+		// This is obviously super fragile (i.e if the error msg wording is updated, this fails)
+		lockFailedErr := "Didn't acquire lock because it is locked and request is configured not to retry."
+		if err.Error() != lockFailedErr {
```

Comment:
That's not idiotmatic. `errors.Is` your friend. https://github.com/loomhq/loomctl/blob/22a7d3d2260c1c8faca3584bc403bca242cbee9c/pkg/lockutil/lock.go#L56-L57

Reply:
That only works if we have an `error`, which we don't have access to because `dynamolock` uses this custom struct `LockNotGrantedError` that doesn't expose the actual error. 😭

Reply:
I don't understand why it would not work. 

In Go an error is an interface, it's not a type. 

```go
Projects/Loom/Repos  production-magenta % go doc builtin.error
package builtin // import "builtin"

type error interface {
	Error() string
}
    
The error built-in interface type is the conventional interface for
 representing an error condition, with the nil value representing no error.
```

Go's interface holds a value and a type, so `error.Is` may operate on underlaying type, which should be `dynamolock.LockNotGrantedError` and this type provides error interface methods: https://github.com/cirello-io/dynamolock/blob/58cfa536df76854d05aad7386c38dab3e723f325/errors.go#L37-L50

https://play.golang.org/p/U8M18s2UbTH

Reply:
@tyuio9 Please show me an example if you think it is possible.

`LockNotGrantedError` is not a specific error type. It is a just a wrapper which means that we need to create our own instance of the error to compare with for `error.Is`. But because the `msg` field is private and there are no public methods to set it, your suggestion is not possible.

https://github.com/cirello-io/dynamolock/blob/master/client.go#L408-L410

Reply:
I opt'd to go with the route that the library uses: https://github.com/cirello-io/dynamolock/blob/master/errors.go#L57-L60